### PR TITLE
dataclients/kubernetes: reject RouteGroups with duplicate backend references

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
For https://github.com/zalando/skipper/issues/2268

### CODE REVIEW

File: dataclients/kubernetes/definitions/routegroups.go [hunk 1]
```diff
@@ -164,6 +166,10 @@ func invalidBackendReference(name string) error {
 	return fmt.Errorf("invalid backend reference: %s", name)
 }
 
+func duplicateBackendReference(name string) error {
```
Comment (szuecs) [inline dataclients/kubernetes/definitions/routegroups.go:169]: I think `namespace/name` would be great

File: dataclients/kubernetes/definitions/testdata/validation/duplicate-default-backend.log [hunk 2]
```diff
@@ -0,0 +1,2 @@
+test-route-group
+duplicate backend reference: app
```
Comment (szuecs) [inline dataclients/kubernetes/definitions/testdata/validation/duplicate-default-backend.log:2]: 2 lines look a bit weird, do I miss something?

File: dataclients/kubernetes/definitions/testdata/validation/route-with-duplicate-backend-reference.log [hunk 3]
```diff
@@ -0,0 +1,2 @@
+test-route-group
+duplicate backend reference: app
```
Comment (szuecs) [inline dataclients/kubernetes/definitions/testdata/validation/route-with-duplicate-backend-reference.log:2]: same as https://github.com/zalando/skipper/pull/2335/files#r1199273710

**Comment by szuecs** [comment]
:+1:

File: dataclients/kubernetes/definitions/testdata/validation/duplicate-default-backend.log (hunk 2 above)
Comment (AlexanderYastrebov) [reply dataclients/kubernetes/definitions/testdata/validation/duplicate-default-backend.log:1]: The fixture test treats each line as a regexp and matches it against actual log lines.
I made it similar to other golden log files but now I removed the first line as we are only interested to check the second one.

**Comment by szuecs** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**